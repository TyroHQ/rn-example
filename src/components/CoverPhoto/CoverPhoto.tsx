import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";
import { Placeholder } from "../Placeholder";

const CoverPhotoContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class CoverPhoto extends React.Component<{}> {
  static defaultProps: CoverPhoto["props"];
  render() {
    return <Placeholder>Cover photo</Placeholder>;
  }
}
