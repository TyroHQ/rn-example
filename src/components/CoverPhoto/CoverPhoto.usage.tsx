import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { CoverPhoto } from "./CoverPhoto";

export const basic = () => <CoverPhoto />;
