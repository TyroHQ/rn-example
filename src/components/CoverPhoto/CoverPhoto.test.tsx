import { shallow } from "enzyme";
import * as usage from "./CoverPhoto.usage";

describe("<CoverPhoto />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
