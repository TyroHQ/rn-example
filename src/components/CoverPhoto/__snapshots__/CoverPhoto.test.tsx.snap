// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoverPhoto /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  Cover photo
</Placeholder>
`;
